index.src.html

```diff
@@ -635,25 +645,29 @@ <h2 id="integration-with-html">Integration with HTML</h2>
   <h2 id="integration-with-css">Integration with CSS</h2>
 
   The CSS Standard does not specify how it fetches resources referenced from
-  stylesheets. However, implementations use the URL a given stylesheet was
-  loaded from as referrer for the requests for resources from that stylesheet.
-
-  Implementations should keep track of a referrer policy for each stylesheet
-  that should be used to create requests for resources from the respective
-  stylesheet.
-
-  For external stylesheets, the referrer policy should be
-  <a>"<code>no-referrer-when-downgrade</code>"</a> unless overwritten by an
-  `<code>Referrer-Policy</code>` header.
-
-  Note: If the stylesheet was loaded via a HTML <{link}> element with a
+  stylesheets. However, implementations should use the <a>location</a> of the
+  <a>stylesheet</a>, <a href="#strip-url">stripped for use as a referrer</a>,
+  as referrer for the requests for resources from that <a>stylesheet</a>,
+  unless the <a>location</a> is <code>null</code>.  The referrer policy should
+  be <a>"<code>no-referrer-when-downgrade</code>"</a> unless overwritten by an
+  `<code>Referrer-Policy</code>` header that was received together with the
+  <a>stylesheet</a>.
+
+  Note: If the <a>stylesheet</a> was loaded via a HTML <{link}> element with a
   declared <{link/referrerpolicy}>, this referrer policy will not affect the
-  requests for resources referenced from the stylesheet.
-
-  For inline stylesheets, and styles applied via an <code>style</code>
-  attribute on an element, the referrer policy is the containing {{Document}}'s
-  referrer policy.  Both the value of the referrer and the value of the
-  referrer policy should be captured at the time a given request is created.
+  requests for resources referenced from the <a>stylesheet</a>.
+
+  If the <a>stylesheet</a>'s <a>location</a> is <code>null</code>, or the
+  request did not originate from a <a>stylesheet</a>, implementations should
+  use the <a>owner node</a>'s <a>node document</a>'s <a for="document">URL</a>,
+  <a href="#strip-url">stripped for use as a referrer</a>, or the
+  <a>element</a>'s <a>node document</a>'s <a for="document">URL</a>, <a
+  href="#strip-url">stripped for use as a referrer</a>, for <a>style
+  elements</a> and <a>style attributes</a> respectively. The referrer policy
+  should be the respective <a>document's referrer policy</a>.
+
+  Both the value of the referrer and the value of the referrer policy should be
+  captured at the time a given request is created.
```

Comment:
I think we can probably clean this up a bit by rearranging things. WDYT of something like:

> ```
> The CSS Standard does not specify how it fetches resources referenced from
> stylesheets. However, implementations should be sure to set the referrer-related
> properties of any <a>requests</a> initiated by stylesheets as follows:
>
> 1.   If a <a>CSS declaration block</a> is responsible for the request,
>      set the <a for="request">referrer</a> to the block's
>      <a for="CSSStyleDeclaration">owner node</a>'s
>      <a>node document</a>'s <a for="document">URL</a>, and the
>      <a for="request">referrer policy</a> to the block's
>      <a for="CSSStyleDeclaration">owner node</a>'s
>      <a>node document</a>'s <a for="Document">referrer policy</a>.
>
> 2.   If a <a>CSS style sheet</a> is responsible for the request,
>      and its <a for="CSSStyleSheet">location</a> is non-null,
>      set the <a for="request">referrer</a> to its
>      <a for="CSSStyleSheet">location</a>, and the <a for="request">referrer
>      policy</a> to its <a for="CSSStyleSheet">referrer policy</a>.
>
>      Issue: This requires that CSS style sheets process `Referrer-Policy`
>      headers, and store a <dfn for="CSSStyleSheet">referrer policy</a>
>      in the same way that <a for="Document" lt="referrer policy">`Document`s
>      do</a>.
>
> 3.  Otherwise, a <a>CSS style sheet</a> with a null location is responsible
>     is responsible for the request: set the <a for="request">referrer</a> to its
>     <a for="CSSStyleSheet">owner node</a>'s
>     <a>node document</a>'s <a for="document">URL</a>, and the
>     <a for="request">referrer policy</a> to the block's
>     <a for="CSSStyleSheet">owner node</a>'s
>     <a>node document</a>'s <a for="Document">referrer policy</a>.
>
> Note: Both the value of the <a>request</a>'s <a for="request">referrer</a> and
> <a for="request">referrer policy</a> are set based on the values at the time
> a given <a>request</a> is created. If a document's referrer policy changes
> during its lifetime, the policy associated with inline stylesheet requests will
> also change.
> ```

WDYT?